maro/simulator/scenarios/data_center/common.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import List
+
+from .virtual_machine import VirtualMachine
+
+
+class Action:
+    """Data center scenario action object, which was used to pass action from agent to business engine.
+
+    Args:
+        vm (VirtualMachine): The virtual machine object, which only contains id, requirement resource, lifetime.
+        pm_id (int): The physical machine id assigned to this vm.
+        buffer_time (int): The remaining buffer time to assign this VM.
+    """
+
+    def __init__(self, assign: bool, vm_req: VirtualMachine, buffer_time: int, pm_id: int = None):
```

Comment:
If we maintain ``be._pending_decision_vm_req``

``assign, vm_id, pm_id`` are enough, and also can prevent the illegal modification